### DISCUSSION TITLE
TLS1.2 client sends device certificates and root certificates

### DISCUSSION THREAD

**Discussion by lubowu** [openssl/openssl#21881] [Q&A]
openssl 1.1.1 version,The TLS1.2 I used the SSL_CTX_load_verify_locations function to load the root certificates.
the client sends device certificates and root certificates by default. How can I disable the sending of root certificates?

**Comment by lubowu** [comment]
How do I set the root certificate not to be sent

**Comment by mattcaswell** [comment]
The function `SSL_CTX_load_verify_locations()` is used for setting the trust store of certificates that the client uses to verify the certificates received from the server.

However, if I understand your question correctly, you are asking about client authentication where the client is sending a certificate to the server? How do you currently set the certificate chain fail in the SSL_CTX?, e.g. do you call a function such as `SSL_CTX_build_cert_chain()` or `SSL_CTX_set0_chain`,  or `SSL_CTX_use_certificate_chain_file()`. The answer to your question will depend on the method that you use to construct/supply the chain to OpenSSL.

**Reply by lubowu** [reply]
Thank you very much for your support.
In TLS1.2 bidirectional authentication, after receiving the device certificate of the server and verifying the certificate, Then the client sends the root certificate and device certificate to the server. However, I do not want the client to send the root certificate, only send the device certificate.
I want the client sends only the device certificate and not the root certificate.
